examples/tcp.rs

```diff
@@ -50,6 +50,57 @@ async fn main() {
         Err(status) => println!("got status during 'add_key' deleted_no_reply: {:?}", status),
     }
 
+    let amount = 1;
+    let increment_key = "increment_key";
+    client
+        .set(increment_key, "0", None, None)
```

Comment:
This doesn't feel right, IMO. We should be setting an integer not a string as the value. Rather we should change the [`set`](https://github.com/Shopify/async-memcached/blob/23c0bd122767647030e2611562891405480c77a5/src/lib.rs#L172) to be more lenient and accept integer values.

Reply:
From a developer experience standpoint I do agree, passing a string literal representation of an integer does feel a bit jank in the context of counters etc.

I've split this out into its own issue and will make that change in a follow-up PR since it doesn't block on the `increment` or `decrement` commands.